TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -936,16 +937,52 @@ public static void CheckStatement(Node node)
 
     public class ProgramChecker
     {
-        public static void OnNode(Program node)
+        public static void OnNode(Program node, TypeCobolOptions typeCobolOptions)
         {
-            node.SetFlag(Node.Flag.MissingEndProgram, !(node.Children.LastOrDefault() is End));
-
-            if (node.IsFlagSet(Node.Flag.MissingEndProgram))
+            //var lastChild = node.Children.LastOrDefault();
+            //if (lastChild is End end)
+            //{
+            //    node.SetFlag(Node.Flag.MissingEndProgram, false);
+            //}
+            //else
+            //{
+            //    node.SetFlag(Node.Flag.MissingEndProgram, true);
+            //    DiagnosticUtils.AddError(node, "\"END PROGRAM\" is missing.", typeCobolOptions.CheckEndProgram.GetMessageCode());
+            //}
+
+            var lastChild = node.Children.LastOrDefault();
+            if (lastChild is End end)
             {
-                DiagnosticUtils.AddError(node,
-                    "\"END PROGRAM\" is missing.", MessageCode.Warning);
+                node.SetFlag(Node.Flag.MissingEndProgram, false);
+                if (typeCobolOptions.CheckEndProgram.IsActive)
+                {
+                    var programEnd = (ProgramEnd)end.CodeElement;
+                    if (programEnd.ProgramName?.Name == null)
+                    {
+                        DiagnosticUtils.AddError(end, "\"PROGRAM END\" should have a program name.", typeCobolOptions.CheckEndProgram.GetMessageCode());
```

Comment:
A vérifier mais cela devrait déjà être remonté par ANTLR.

Reply:
ANTLR produces the error: `Syntax error : mismatched input '.' expecting {alphanumeric literal, hexadecimal alphanumeric literal, null terminated alphanumeric literal, user defined word}`
While this produces the error like: `Warning: "PROGRAM END" should have a program name. "Nested0" was assumed.` which is more expressive according to me. It is also closer to the IBM compiler.
Should we keep this error or remove it?